src/mlpack/methods/edge_boxes/feature_extraction.hpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file feature_extraction.hpp
+ * @author Nilay Jain 
+ *
+ * Feature Extraction for the edge_boxes algorithm.
+ */
+#ifndef MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_HPP
+#define MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_HPP
+#define INF 999999.9999
+#define EPS 1E-20
+#include <mlpack/core.hpp>
```

Comment:
a : If possible, please do not use macro/pre-processor, they have many defects.

1 : they do not respect scope. 
2 : debug message is hard to read
3 : Not type safe

So, do we have a better choice?Yes, we have
it is constexpr, this keyword tell the compiler, I know this const value at compile time
so you could declare it as 

```
constexpr double inf = 999999.9999;
constexpr double eps = 1e-20;
```
constexpr respect scope, compiler will give you better error message rather than show you "1e-20" or "99999.99999"

b : If possible, could you place the eps in your class rather than put it in global namespace?

c : c++11 provide [standard infinity value f](http://en.cppreference.com/w/cpp/types/numeric_limits/infinity)or us

`double inf = std::numeric_limits<double>::infinity();`

Very cool, right?